### ISSUE TITLE
Trying to use strict BEM but scss-lint complains

### ISSUE THREAD

**Issue by hendrikeng** [brigade/scss-lint#859]
.block-name__element-name_modifier-name doesnt work ..i am bit confused 

.block-name__element-name -> works
.block-name_modifier-name -> works 
.block-name__element-name_modifier-name -> doesnt 

am i missing something ? thanks for your time

**Comment by abea** [comment]
Have you tried `.block-name__element-name--modifier-name`? That's how I understand the syntax to work.

**Comment by hendrikeng** [comment]
doesnt work either but according to https://en.bem.info the syntax i tried is correct .

> .block-name__element-name_modifier-name

**Comment by sds** [comment]
I am not familiar enough with the BEM methodology to weigh in here. Would love a pull request fixing this issue if there is indeed a problem. Thanks!

**Comment by hendrikeng** [comment]
@sds sorry i don't know how do fix it via a pull request, i just stumbled across it.

**Comment by widoz** [comment]
There are two syntax for the bem convention, one of which is the @abea reply. 
So, the lint should support both.

**Comment by ryanlloydfrench** [comment]
Keen to bump this issue.

Currently this is breaking for me when using the `strict_BEM` convention. Writing BEM but the linter is returning warnings. Seems to be when using the double hyphen for modifier. 

See examples below:

`.box__heading` _**works**_
`.box--large` _**warning**_
`.box__heading--large` _**warning**_

**Comment by trotzig** [comment]
@ryanlloydfrench Judging by your example, I believe you should switch to `hyphenated_BEM` (which is that other syntax @widoz is referring to). According to "strict" BEM, [modifiers use a single underscore](https://en.bem.info/methodology/naming-convention/#modifier-name).

**Comment by joelmoss** [comment]
Ok, so I am also coming across the issue when using the `strict_BEM` convention. The following fails the linting:

```css
.nav__link_active {
  color: #fff;
}
```

The above is a valid BEM strict syntax, but scss-lint raises a warning.

The [tests](https://github.com/brigade/scss-lint/blob/master/spec/scss_lint/linter/selector_format_spec.rb#L520-L524) also seem to support the above as being _invalid_, but it is valid.
